runtime/v2/runc/task/service.go

```diff
@@ -158,10 +160,14 @@ func (s *service) Start(ctx context.Context, r *taskAPI.StartRequest) (*taskAPI.
 	}
 
 	// hold the send lock so that the start events are sent before any exit events in the error case
-	s.eventSendMu.Lock()
+	lockKey := r.ExecID
+	if lockKey == "" {
+		lockKey = r.ID
+	}
```

Comment:
It looks like `r.ExecID` only needs to be unique for a given `r.ID` so I'm pretty sure the map key would need to be the tuple `(r.ID, r.ExecID)`.
```go
// Naming things is hard. Try to think of a better name than this.
type processEventsKey struct { ID, ExecID string }

s.processEventsLocker.Store(processEventsKey{ID: r.ID, ExecID: r.ExecID}, ...)
```

Alternatively, have a separate map for each container, e.g. by moving it to the `runc.Container` struct?